csharp/src/Ice/Properties.cs

```diff
@@ -54,7 +181,17 @@ public interface Properties
     ///  </param>
     /// <returns>The property value interpreted as an integer, or the default value.
     ///  </returns>
-    int getIcePropertyAsInt(string key);
+    public int getIcePropertyAsInt(string key)
+    {
+        string defaultValueString = getDefaultProperty(key);
+        int defaultValue = 0;
+        if (!string.IsNullOrEmpty(defaultValueString))
```

Comment:
Since this getDefaultProperty is a new API, it should be "nullable correct".  Here, you write it can't be null.